Uncaught exception thrown when only 1 of schemas/schematrons are provided via command-line

### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

I installed the Windows version of validate_3.3.1 today.

(1) It appears that when I try to validate a PDS4 product, validate throws errors.  See below.

(2) when I run this command:  validate -V or validate --help
It works.  So, I am guessing no issues with the installation.

(3) when I run this command:   validate -t 227RI_PROPRETRG001_IS____680_F1_039R_edited.xml >> log.txt

This is displayed:
```
D:\WINWORD\Data_Prep_HandBook\aaaVer_9_20130225\schemas\dictionary_CASSINI_mission\ATMOS_test_20231113> validate331 227RI_PROPRETRG001_IS____680_F1_039R.xml >> log.txt
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:64)
        at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
        at gov.nasa.pds.tools.validate.task.ValidationTask.execute(ValidationTask.java:130)
        at gov.nasa.pds.tools.validate.task.BlockingTaskManager.submit(BlockingTaskManager.java:26)
        at gov.nasa.pds.tools.label.LocationValidator.validate(LocationValidator.java:285)
        at gov.nasa.pds.validate.ValidateLauncher.doValidation(ValidateLauncher.java:1454)
        at gov.nasa.pds.validate.ValidateLauncher.processMain(ValidateLauncher.java:1740)
        at gov.nasa.pds.validate.ValidateLauncher.main(ValidateLauncher.java:1784)
Caused by: java.lang.NullPointerException
        at gov.nasa.pds.tools.validate.TargetExaminer.isTargetBundleType(TargetExaminer.java:59)
        at gov.nasa.pds.tools.util.Utility.getTargetType(Utility.java:301)
        at gov.nasa.pds.tools.validate.ValidationTarget.<init>(ValidationTarget.java:47)
        at gov.nasa.pds.tools.validate.ValidationTarget.<init>(ValidationTarget.java:41)
        at gov.nasa.pds.tools.util.Utility.getValidationTarget(Utility.java:88)
        at gov.nasa.pds.tools.validate.ValidationProblem.<init>(ValidationProblem.java:69)
        at gov.nasa.pds.tools.label.CachedLSResourceResolver.resolveResource(CachedLSResourceResolver.java:207)
        at gov.nasa.pds.tools.label.XMLCatalogResolver.resolveResource(XMLCatalogResolver.java:381)
        at gov.nasa.pds.tools.validate.rule.pds4.LabelValidationRule.loadSingleSchemaIntoSources(LabelValidationRule.java:397)
        at gov.nasa.pds.tools.validate.rule.pds4.LabelValidationRule.validateLabelSchemas(LabelValidationRule.java:572)
        at gov.nasa.pds.tools.validate.rule.pds4.LabelValidationRule.validateLabel(LabelValidationRule.java:250)
        ... 12 more
```

### 🕵️ Expected behavior

I expected validate to validate the product using this command:
 >> validate -t 227RI_PROPRETRG001_IS____680_F1_039R_edited.xml >> log.txt

### 📜 To Reproduce

1. unzip files
2. validate -t 227RI_PROPRETRG001_IS____680_F1_039R_edited.xml >> log.txt
3. review log.txt
...


### 🖥 Environment Info

- Version of this software: validate_3.3.1
- Operating System: Windows
...


### 📚 Version of Software Used

validate_3.3.1 for Windows.  I installed using the zip.

### 🩺 Test Data / Additional context

[ATMOS_test_20231113.zip](https://github.com/NASA-PDS/validate/files/13368957/ATMOS_test_20231113.zip)


### 🦄 Related requirements
- 🦄 https://github.com/NASA-PDS/validate/issues/599

### ⚙️ Engineering Details

--- a/src/main/java/gov/nasa/pds/tools/validate/TargetExaminer.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/TargetExaminer.java
@@ -56,6 +56,9 @@ public class TargetExaminer extends Target {
     boolean targetIsBundleFlag = false;
 
     // Do a sanity check if the file or directory exist.
+    if (!"file".equalsIgnoreCase(url.getProtocol())) {
+      return (targetIsBundleFlag);     
+    }
     if (!FileUtils.toFile(url).exists()) {
       LOG.error("Provided file does not exist: {}", FileUtils.toFile(url));
       return (targetIsBundleFlag);
@@ -85,6 +88,9 @@ public class TargetExaminer extends Target {
     boolean targetIsCollectionFlag = false;
 
     // Do a sanity check if the file or directory exist.
+    if (!"file".equalsIgnoreCase(url.getProtocol())) {
+      return (targetIsCollectionFlag);     
+    }
     if (!FileUtils.toFile(url).exists()) {
       LOG.error("Provided file does not exist: {}", FileUtils.toFile(url));
       return (targetIsCollectionFlag);

--- a/src/main/java/gov/nasa/pds/tools/validate/content/array/ArrayContentValidator.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/content/array/ArrayContentValidator.java
@@ -300,7 +300,7 @@ public class ArrayContentValidator {
     // {},{},{}",dataType,value,rangeChecker.contains(value));
 
     if (!isSpecialConstant) {
-      if (!rangeChecker.contains(value)) {
+      if (!rangeChecker.contains(value) && !isNanOrInf(value)) {
         addArrayProblem(ExceptionType.ERROR, ProblemType.ARRAY_VALUE_OUT_OF_DATA_TYPE_RANGE,
             ArrayContentValidator.tableNameReportStr
                 + "Value is not within the valid range of the data type '" + dataType.name() + "': "
@@ -320,6 +320,17 @@ public class ArrayContentValidator {
     }
   }
 
+  private boolean isNanOrInf (Number value) {
+    if (value instanceof Double) {
+      Double v = (Double)value;
+      return v.isInfinite() || v.isNaN();
+    }
+    if (value instanceof Float) {
+      Float v = (Float)value;
+      return v.isInfinite() || v.isNaN();
+    }
+   return false;
+  }
   private static boolean sameContent (Number number, String constant_repr) {
     if (constant_repr == null) return false;
     if (number.toString().equals(constant_repr)) {

--- a/src/test/resources/features/developer.feature
+++ b/src/test/resources/features/developer.feature
@@ -99,7 +99,7 @@ Scenario Outline: Execute validate command for tests below.
 # fixed: |"NASA-PDS/validate#529 Fail Update Compare Scale" | "github529" | 1 | "1 errors expected" | "ARRAY_VALUE_OUT_OF_MIN_MAX_RANGE" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github529_2.json -s json -t {resourceDir}/github529/fail/m0126360726f4_2p_cif_gbl.xml " | "report_github529_2.json" |
 
 # Validate#514 - binary tables should allow NaN
-|"NASA-PDS/validate#514 Success Update Compare Scale" | "github514" | 3 | "3 errors expected" | "ARRAY_VALUE_OUT_OF_DATA_TYPE_RANGE" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github514_1.json -s json -t {resourceDir}/github514/success/8array.xml" | "report_github514_1.json" |
+|"NASA-PDS/validate#514 Success Update Compare Scale" | "github514" | 0 | "0 errors expected" | "totalErrors" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github514_1.json -s json -t {resourceDir}/github514/success/8array.xml" | "report_github514_1.json" |
 
 # Validate#499
 |"NASA-PDS/validate#499 Success Table EOL" | "github499" | 0 | "0 errors expected" | "totalErrors" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github499_1.json -s json -t {resourceDir}/github499/success/M7_217_044546_N.xml" | "report_github499_1.json" |

